### DISCUSSION TITLE
How to get all form inputs n a single object ?

### DISCUSSION THREAD

**Discussion by neomn** [fastapi/fastapi#10559] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Form

app = FastAPI()


@app.post("/login/")
async def login(username: Annotated[str, Form()], password: Annotated[str, Form()]):
    return {"username": username}
```

### Description

i have a form that has lots of inputs , it's a mess if i want to get them one by one , any way to get them all in a single object ? 
after getting them in a single object i can then validate them in a separated pydantic class , much cleaner syntax, right? 

### Operating System

Linux

### FastAPI Version

0.100.0

### Pydantic Version

2

### Python Version

3.11

**Comment by dfalt974** [comment]
convert Form to Dict
data= await request.form()
data = data._dict

**Comment by YuriiMotov** [comment, marked as answer]
Since FastAPI 0.113.0 you can use Pydantic models to define `Form` parameters 🎉

Docs: https://fastapi.tiangolo.com/tutorial/request-form-models/
